backend/app/anti_recommenders/anti_recommender_proxy.py

```diff
@@ -0,0 +1,32 @@
+import os
+from typing import Tuple 
+from .anti_recommender import AntiRecommender
+from .open_ai.regular_open_ai_anti_recommender import RegularOpenAIAntiRecommender
+
+class AntiRecommenderProxy(AntiRecommender):
+    """
+    The Proxy class for the anti-recommendation engine.
+    """
+
+    def __init__(self, wikipedia_title: str = None):
+        self.wikipedia_title : str = wikipedia_title
+        self.anti_recommender = None
+
+    def set_title(self, title:str) -> None:
+        """store the title of the Wikipedia article."""
+
+        self.wikipedia_title = title 
+
+    def get_title(self) -> str | None:
+        """return the title of the Wikipedia article."""
+        
+        return self.wikipedia_title 
+    
+    def generate_anti_recommendations(self, type: str = "openai") -> Tuple[Tuple, ...]:
```

Comment:
ABC should have same signature.

Ruff will also tell you to use a better signature here, `tuple[tuple[actual type, ...], ...]`.

Reply:
Unclear why `type` is passed to generate, seems like something that'd be figured out in DI / as a constructor parameter.

Reply:
I have removed `type` from `generate_anti_recommendations()` and made it a constructor parameter.